File: c/armstrong-numbers/src/armstrong_numbers.c
#include "armstrong_numbers.h"
#include <math.h>


int is_armstrong_number(int candidate)
{
	int result;
	int control;
	double count;

	count = ceil(log10(candidate));

	result = 0;
	control = candidate;
	while (control) {
		result += (int) pow((double) (control % 10), count);
		control /= 10;
	}

	return (result == candidate) ? 1 : 0;
}

File: python/armstrong-numbers/armstrong_numbers.py
import math


def is_armstrong_number(number):
    if number:
        count = math.ceil(math.log10(number))

    result = 0
    calc_num = number
    while calc_num:
        result += (calc_num % 10) ** count
        calc_num = calc_num // 10

    return result == number
